### PR TITLE
planner: fix begin statement consumed read_ts wrongly

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?
The following behavior is not expected.
```sql
mysql> set transaction read only as of timestamp '2021-06-16 11:20:00';
Query OK, 0 rows affected (0.05 sec)

mysql> delete from t1 where id=3;
ERROR 1105 (HY000): only support read-only statement during read-only staleness transactions
mysql> commit;
Query OK, 0 rows affected (0.05 sec)

mysql> start transaction read only as of timestamp now()-interval 1 minute;
ERROR 1105 (HY000): start transaction read only as of is forbidden after set transaction read only as of
mysql> start transaction read only as of timestamp now()-interval 1 minute;
Query OK, 0 rows affected (0.05 sec)
```

### What is changed and how it works?

Fix that the read_ts would be consumed wrongly.

### Check List 

Tests 

- Unit test

### Release note 

* No release note

### CODE REVIEW

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -465,6 +465,13 @@ func (s *testStaleTxnSerialSuite) TestSetTransactionReadOnlyAsOf(c *C) {
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Equals, "start transaction read only as of is forbidden after set transaction read only as of")
 	tk.MustExec(`SET TRANSACTION READ ONLY as of timestamp '2021-04-21 00:42:12'`)
+	err = tk.ExecToErr(`START TRANSACTION READ ONLY AS OF TIMESTAMP '2020-09-06 00:00:00'`)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "start transaction read only as of is forbidden after set transaction read only as of")
+
+	tk.MustExec("begin")
+	tk.MustExec("commit")
```
Comment (xhebox) [inline executor/stale_txn_test.go:473]: Assertion on `startTS`?

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: executor/stale_txn_test.go (hunk 1 above)
Comment (Yisaer) [reply executor/stale_txn_test.go:474]: updated.

**Review by xhebox — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**